7ef76925d9c19ef74874e1735e2436e56d0c4897 (from 6b14d7faf0bad026a81a27bac07b47691f621b8f)
diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ INSTALL=install
 
 SCRIPTS=git-apply-patch-script git-merge-one-file-script git-prune-script \
 	git-pull-script git-tag-script git-resolve-script git-whatchanged \
-	git-deltafy-script
+	git-deltafy-script git-fetch-script
 
 PROG=   git-update-index git-diff-files git-init-db git-write-tree \
 	git-read-tree git-commit-tree git-cat-file git-fsck-cache \
diff --git a/git-pull-script b/git-fetch-script
similarity index 87%
copy from git-pull-script
copy to git-fetch-script
--- a/git-pull-script
+++ b/git-fetch-script
@@ -39,8 +39,3 @@ download_one "$merge_repo/$merge_name" "
 
 echo "Getting object database"
 download_objects "$merge_repo" "$(cat "$GIT_DIR"/MERGE_HEAD)"
-
-git-resolve-script \
-	"$(cat "$GIT_DIR"/HEAD)" \
-	"$(cat "$GIT_DIR"/MERGE_HEAD)" \
-	"$merge_repo"
diff --git a/git-pull-script b/git-pull-script
--- a/git-pull-script
+++ b/git-pull-script
@@ -6,39 +6,7 @@ merge_name=${2:-HEAD}
 : ${GIT_DIR=.git}
 : ${GIT_OBJECT_DIRECTORY="${SHA1_FILE_DIRECTORY-"$GIT_DIR/objects"}"}
 
-download_one () {
-	# remote_path="$1" local_file="$2"
-	case "$1" in
-	http://*)
-		wget -q -O "$2" "$1" ;;
-	/*)
-		test -f "$1" && cat >"$2" "$1" ;;
-	*)
-		rsync -L "$1" "$2" ;;
-	esac
-}
-
-download_objects () {
-	# remote_repo="$1" head_sha1="$2"
-	case "$1" in
-	http://*)
-		git-http-pull -a "$2" "$1/"
-		;;
-	/*)
-		git-local-pull -l -a "$2" "$1/"
-		;;
-	*)
-		rsync -avz --ignore-existing \
-			"$1/objects/." "$GIT_OBJECT_DIRECTORY"/.
-		;;
-	esac
-}
-
-echo "Getting remote $merge_name"
-download_one "$merge_repo/$merge_name" "$GIT_DIR"/MERGE_HEAD
-
-echo "Getting object database"
-download_objects "$merge_repo" "$(cat "$GIT_DIR"/MERGE_HEAD)"
+git-fetch-script "$merge_repo" "$merge_name"
 
 git-resolve-script \
 	"$(cat "$GIT_DIR"/HEAD)" \
