diff a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ INSTALL=install
 
 SCRIPTS=but-apply-patch-script but-merge-one-file-script but-prune-script \
 	but-pull-script but-tag-script but-resolve-script but-whatchanged \
-	but-deltafy-script
+	but-deltafy-script but-fetch-script
 
 PROG=   but-update-index but-diff-files but-init-db but-write-tree \
 	but-read-tree but-cummit-tree but-cat-file but-fsck-cache \
diff a/but-fetch-script b/but-fetch-script
--- /dev/null
+++ b/but-fetch-script
@@ -0,0 +1,41 @@
+#!/bin/sh
+#
+merge_repo=$1
+merge_name=${2:-HEAD}
+
+: ${BUT_DIR=.but}
+: ${BUT_OBJECT_DIRECTORY="${SHA1_FILE_DIRECTORY-"$BUT_DIR/objects"}"}
+
+download_one () {
+	# remote_path="$1" local_file="$2"
+	case "$1" in
+	http://*)
+		wget -q -O "$2" "$1" ;;
+	/*)
+		test -f "$1" && cat >"$2" "$1" ;;
+	*)
+		rsync -L "$1" "$2" ;;
+	esac
+}
+
+download_objects () {
+	# remote_repo="$1" head_sha1="$2"
+	case "$1" in
+	http://*)
+		but-http-pull -a "$2" "$1/"
+		;;
+	/*)
+		but-local-pull -l -a "$2" "$1/"
+		;;
+	*)
+		rsync -avz --ignore-existing \
+			"$1/objects/." "$BUT_OBJECT_DIRECTORY"/.
+		;;
+	esac
+}
+
+echo "Getting remote $merge_name"
+download_one "$merge_repo/$merge_name" "$BUT_DIR"/MERGE_HEAD
+
+echo "Getting object database"
+download_objects "$merge_repo" "$(cat "$BUT_DIR"/MERGE_HEAD)"
diff a/but-pull-script b/but-pull-script
--- a/but-pull-script
+++ b/but-pull-script
@@ -6,39 +6,7 @@ merge_name=${2:-HEAD}
 : ${BUT_DIR=.but}
 : ${BUT_OBJECT_DIRECTORY="${SHA1_FILE_DIRECTORY-"$BUT_DIR/objects"}"}
 
-download_one () {
-	# remote_path="$1" local_file="$2"
-	case "$1" in
-	http://*)
-		wget -q -O "$2" "$1" ;;
-	/*)
-		test -f "$1" && cat >"$2" "$1" ;;
-	*)
-		rsync -L "$1" "$2" ;;
-	esac
-}
-
-download_objects () {
-	# remote_repo="$1" head_sha1="$2"
-	case "$1" in
-	http://*)
-		but-http-pull -a "$2" "$1/"
-		;;
-	/*)
-		but-local-pull -l -a "$2" "$1/"
-		;;
-	*)
-		rsync -avz --ignore-existing \
-			"$1/objects/." "$BUT_OBJECT_DIRECTORY"/.
-		;;
-	esac
-}
-
-echo "Getting remote $merge_name"
-download_one "$merge_repo/$merge_name" "$BUT_DIR"/MERGE_HEAD
-
-echo "Getting object database"
-download_objects "$merge_repo" "$(cat "$BUT_DIR"/MERGE_HEAD)"
+but-fetch-script "$merge_repo" "$merge_name"
 
 but-resolve-script \
 	"$(cat "$BUT_DIR"/HEAD)" \
